### ISSUE TITLE
[Flaky test] Conformance - GCE - master - kubetest2

### ISSUE THREAD

**Issue by mehabhalodiya** [kubernetes/kubernetes#118928]
### Which jobs are failing?

master-blocking
- Conformance - GCE - master - kubetest2

### Which tests are failing?

- Kubernetes e2e suite.[It] [sig-apps] Daemon set [Serial] should rollback without unnecessary restarts [Conformance] [Changes](https://github.com/kubernetes/kubernetes/compare/3c380199e...470889278?)
- ci-kubernetes-gce-conformance-latest-kubetest2.Overall [Changes](https://github.com/kubernetes/kubernetes/compare/3c380199e...470889278?)
- kubetest2.Test [Changes](https://github.com/kubernetes/kubernetes/compare/3c380199e...470889278?)

![Screenshot from 2023-06-28 15-01-46](https://github.com/kubernetes/kubernetes/assets/73488906/21dc26d7-62f9-4772-9b4c-817097a6a5bc)

### Since when has it been failing?

2023-06-27 16:43 PDT

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#Conformance%20-%20GCE%20-%20master%20-%20kubetest2

### Reason for failure (if possible)

Kubernetes e2e suite: [It] [sig-apps] Daemon set [Serial] should rollback without unnecessary restarts [Conformance] 
```
{ failed [FAILED] unexpected pod daemon-set-lmdd2 be restarted In [It] at: test/e2e/apps/daemon_set.go:500 @ 06/28/23 07:35:29.181 }
```

### Relevant SIG(s)

/sig cloud-provider
/sig apps
cc @kubernetes/ci-signal

**Comment by aojea** [comment]
This is more critical than I expected it starts to fail in other jobs https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=should%20rollback%20without%20unnecessary%20restarts

https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20master
https://testgrid.k8s.io/sig-network-kind#sig-network-kind,%20ipvs,%20dual,%20master

From the diff https://github.com/kubernetes/kubernetes/compare/16fe4a4cc...3c380199e?

the best candidate seems https://github.com/kubernetes/kubernetes/pull/118716

@alculquicondor @soltysh can you PTAL

/priority critical-urgent

**Comment by mehabhalodiya** [comment]
@aojea @alculquicondor @soltysh anyone from you can please add `/triage accepted` to this issue?

**Comment by alculquicondor** [comment]
I have a proposed fix in #118935.
Is there a job I can add?
